internal/models/permission_record.go

```diff
@@ -0,0 +1,29 @@
+package models
+
+import (
+	"time"
+)
+
+// PermissionRecord represents a SACD permission record
+type PermissionRecord struct {
+	SpecVersion string    `json:"specversion"`
+	Timestamp   time.Time `json:"timestamp"`
+	Type        string    `json:"type"`
+	Data        struct {
+		Grantor struct {
+			Address string `json:"address"`
+		} `json:"grantor"`
+		Grantee struct {
+			Address string `json:"address"`
+		} `json:"grantee"`
+		EffectiveAt time.Time `json:"effectiveAt"`
+		ExpiresAt   time.Time `json:"expiresAt"`
+		Agreements  []struct {
+			Type        string `json:"type"`
+			Asset       string `json:"asset"`
+			Permissions []struct {
+				Name string `json:"name"`
+			} `json:"permissions"`
+		} `json:"agreements"`
+	} `json:"data"`
+}
```

Comment:
Break this up to something like this so we are not using nested structs
```suggestion
// PermissionRecord is the top-level structure for permission events
type PermissionRecord struct {
	SpecVersion string         `json:"specversion"`
	Timestamp   time.Time      `json:"timestamp"`
	Type        string         `json:"type"`
	Data        PermissionData `json:"data"`
}

// PermissionData contains the core permission data
type PermissionData struct {
	Grantor     Address     `json:"grantor"`
	Grantee     Address     `json:"grantee"`
	EffectiveAt time.Time   `json:"effectiveAt"`
	ExpiresAt   time.Time   `json:"expiresAt"`
	Agreements  []Agreement `json:"agreements"`
}

// Agreement represents a permission agreement for an asset
type Agreement struct {
	Type        string       `json:"type"`
	Asset       string       `json:"asset"`
	Permissions []Permission `json:"permissions"`
}

// Permission defines a single permission
type Permission struct {
	Name string `json:"name"`
}

// Address represents a blockchain address
type Address struct {
	Address string `json:"address"`
}
```